### PR TITLE
Limit v1 webhooks to None and NoneOnDryRun side effects classes

### LINKED ISSUE #68839: Change default for SideEffect in admissionregistration Webhook v1
As described in [dry-run KEP](https://github.com/kubernetes/community/blob/master/keps/sig-api-machinery/0015-dry-run.md#admission-controllers) and [that TODO](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/admissionregistration/v1beta1/defaults.go#L39), the value of SideEffect in the webhook registration should eventually change to "NoneOnDryRun":

> This field will be defaulted to true and deprecated in v1, and completely removed in v2. All webhooks registered with v2 will be assumed to support dry run.

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind api-change
/kind feature

**What this PR does / why we need it**:
Requires admission webhooks registered via v1 to not have side effects (only `None` and `NoneOnDryRun` are allowed as side effects classes)

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/kubernetes/issues/68839

**Does this PR introduce a user-facing change?**:
(will add to v1 promotion PR release note in https://github.com/kubernetes/kubernetes/pull/79549)
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/api/admissionregistration/v1/types.go [hunk 1]
```diff
@@ -273,8 +273,8 @@ type ValidatingWebhook struct {
 	// +optional
 	ObjectSelector *metav1.LabelSelector `json:"objectSelector,omitempty" protobuf:"bytes,10,opt,name=objectSelector"`
 
-	// SideEffects states whether this webhookk has side effects.
-	// Acceptable values are: Unknown, None, Some, NoneOnDryRun
+	// SideEffects states whether this webhook has side effects.
```
Comment (apelisse) [inline staging/src/k8s.io/api/admissionregistration/v1/types.go]: Would it be worth clarifying what happens when not specified (can it, and should we default it)?

Comment (liggitt) [reply staging/src/k8s.io/api/admissionregistration/v1/types.go]: it cannot be omitted in v1. it's required and not defaulted

Comment (apelisse) [reply staging/src/k8s.io/api/admissionregistration/v1/types.go]: AFAIU, the distinction between `None` and `NoneOnDryRun` makes no difference, so I'm not really sure why we force people to specify a value that adds no value?

Comment (liggitt) [reply staging/src/k8s.io/api/admissionregistration/v1/types.go]: making it explicit in v1 ensures all manifests ack that no side effects are allowed. the distinction between None and NoneOnDryRun doesn't matter for the dispatcher, but allows expressing that there is a difference in behavior specifically for dry run

**Pull request merged**